nipype/interfaces/traits_extension.py

```diff
@@ -218,6 +218,38 @@ def __init__(self, value='', auto_set=False, entries=0,
         super(Directory, self).__init__(value, auto_set, entries, exists,
                                         **metadata)
 
+class ImageFile(File):
+    """ Defines a trait of specific neuroimaging files """
+
+    def __init__(self, value='', filter=None, auto_set=False, entries=0,
+                 exists=False, types=['nii', 'hdr', 'img'], compressed=True,
```

Comment:
say one of the other interfaces (e.g., dcm2niix) were to restrict input files to dicoms. how would you determine that the input was valid for the MGH dicoms for example, which don't have extensions?

i'm also thinking of how we can use this interface to auto-(un)compress eventually instead of restricting it.

further compressed nifti pair would involve img/hdr.gz pairs. if we check only the header, then spm could still run into trouble.

so types can be format with details about compressed and uncompressed being something like this to start with:

```python
{'nifti1': [('.nii', '.nii.gz'), (('.hdr', '.img'), ('.hdr', '.img.gz'))],
 'mgh': [('.mgh', '.mgz')], 
  ...
}
```

at least for now, since this solution is focused on spm, let's use Nifti1 as the type and check for the various combinations listed above.